src/video-player.js

```diff
@@ -118,16 +118,26 @@ class VideoPlayer extends Utils.mixin(Eventable) {
     this._initPlaylistWidget();
     this._initJumpButtons();
     this._setVideoJsListeners(ready);
-    this._sendAnalytics(this.playerOptions);
   }
 
-  _sendAnalytics(options) {
+  _sendInternalAnalytics(additionalOptions = {}) {
+    if (this._internalAnalyticsSent) {
+      return;
+    }
+
     try {
+      const cloudName = this.cloudinaryConfig().cloud_name;
```

Comment:
Why not do it together with everything else inside `getCloudinaryOptions`? this seems to unnecessarily complicate this.

Reply:
do you mean to put baseParams into `getAnalyticsFromPlayerOptions` ? If so it would mean we are expanding responsibility of this function to have access to video player instance to get those data - or we would simply pass it there which IMO is similar to what is there

Reply:
I changed it to get cloud name directly `cloudName: options.cloudinary.cloudinaryConfig.cloud_name` but we need to keep it separated because of internal fn `getInstanceId`